CleverTapSDK/InApps/CTInAppDisplayManager.m

```diff
@@ -156,6 +156,9 @@ - (void)setInAppNotificationDelegate:(id <CleverTapInAppNotificationDelegate>)de
 - (void)_addInAppNotificationsToQueue:(NSArray *)inappNotifs {
     @try {
         NSArray *filteredInAppNotifs = [self filterNonRegisteredTemplates:inappNotifs];
+        if (pushPrimerManager.pushPermissionStatus == CTPushEnabled) {
```

Comment:
On Android, the in-app is filtered out in `displayNotification` method, which is the one before displaying the in-app. Please sync with Vassil and consider adding a check in the `displayNotification` method. This one could probably stay as an optimization.

Reply:
I have checked this with Vassil, he mentioned that we can do it while adding on queue also. He has added before showing in-app. I have pinged you the optimisation to add a check on display notification also, if it looks good let me know I will update this.

Reply:
I have added safety check inside display notification as well.